Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,413 @@
+                    return false; // input collider is invalid
+                }
+            }
+
+            m_bakedConfinerResolutionCache = m_bakedConfinerResolution;
+            m_sensorRatioCache = sensorRatio;
+            GetConfinerOven().BakeConfiner(m_originalPath, m_sensorRatioCache, m_bakedConfinerResolutionCache, 
+                m_MaxOrthoSize, m_ShrinkToPointsExperimental);
+            m_confinerStates = GetConfinerOven().GetShrinkablePolygonsAsConfinerStates();
+
+            m_boundingShapePositionCache = m_BoundingShape2D.transform.position;
+            m_boundingShapeRotationCache = m_BoundingShape2D.transform.rotation;
+            m_boundingShapeScaleCache = m_BoundingShape2D.transform.localScale;
+
+            BakeProgress = BakeProgressEnum.BAKED;
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if the input bounding shape was moved, rotated, or scaled.
+        /// </summary>
+        /// <returns></returns>
+        private bool BoundingShapeTransformChanged()
+        {
+            return m_BoundingShape2D != null && 
+                   (m_boundingShapePositionCache != m_BoundingShape2D.transform.position ||
+                    m_boundingShapeScaleCache != m_BoundingShape2D.transform.localScale ||
+                    m_boundingShapeRotationCache != m_BoundingShape2D.transform.rotation);
+        }
+        
+        private ConfinerOven.ConfinerState m_confinerCache;
+        /// <summary>
+        /// Check that the path cache was converted from the current confiner cache, or
+        /// converts it if the frustum height was changed.
+        /// </summary>
+        /// <param name="confinerStateChanged">Confiner cache was changed</param>
+        /// <param name="frustumHeight">Camera frustum height</param>
+        private void ValidatePathCache(bool confinerStateChanged, float frustumHeight)
+        {
+            if (confinerStateChanged ||
+                m_currentPathCache == null || 
+                Math.Abs(frustumHeight - m_frustumHeightCache) > m_bakedConfinerResolution)
+            {
+                m_frustumHeightCache = frustumHeight;
+                m_confinerCache = GetConfinerOven().GetConfinerAtFrustumHeight(m_frustumHeightCache);
+                ShrinkablePolygon.ConvertToPath(m_confinerCache.polygons, m_frustumHeightCache, out m_currentPathCache);
+            }
+        }
+        
+        /// <summary>
+        /// Singleton for this object's ConfinerOven.
+        /// </summary>
+        /// <returns>ConfinerOven</returns>
+        private ConfinerOven GetConfinerOven()
+        {
+            if (m_confinerBaker == null)
+            {
+                m_confinerBaker = new ConfinerOven();
+            }
+
+            return m_confinerBaker;
+        }
+        
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            ForceBake();
+        }
+        
+        private void OnDrawGizmos()
```

Comment:
Do you need to handle confiner offset, as you did in https://github.com/Unity-Technologies/upm-package-cinemachine/pull/108